### PR TITLE
chore(angular-boiler-plate):add changes to angular boilerplate and updated readme

### PR DESCRIPTION
**Pull request by Deepika516**
## Description

- Documentation of Angular-Boilerplate
- added readme for auth,core,theme and shared module.
-  Made changes to code:
  Removed breadcrumbs,cell renderer,warning dialog and grid utils from shared module.
  Replaced select and select-multiple with bizbook- select.

### CODE REVIEW

File: apps/boiler-plate/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# AngularBoilerPlate
+<!-- # AngularBoilerPlate
 
 This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 15.0.4.
```
Comment (yeshamavani) [inline apps/boiler-plate/README.md:3]: cli version

File: apps/boiler-plate/README.md [hunk 2]
```diff
@@ -24,4 +24,100 @@ Run `ng e2e` to execute the end-to-end tests via a platform of your choice. To u
...
+
+app.module.ts: This file defines the main module of your application.
+
+shared/: This directory contains shared components, services, and modules for your application.
+
+feature/: This directory contains feature-specific components, services, and modules for your application.
+
+# Project Features
+
+authComponent: this component authenticates the user and receives an access token from Auth
```
Comment (yeshamavani) [inline apps/boiler-plate/README.md:88]: we also provide various basic commands, Adapters -- anyAdapter, theme, localization service, mention the decorators, interceptors, store, toaster, env-resolver, guards, basic commands, auth specific commands, main module with a home module.
Also shared components like breakcrum, grid, gantt etc 
add others if I missed anything.

File: apps/boiler-plate/README.md [hunk 3]
```diff
@@ -24,4 +24,100 @@ Run `ng e2e` to execute the end-to-end tests via a platform of your choice. To u
...
+
+loginComponent The login component uses the authentication service to login to the application. If the user is already logged in they are automatically redirected to the home page.
+
+homeComponent The home page shows the content what is exactly going into the project
+
+# Usage
+
```
Comment (yeshamavani) [inline apps/boiler-plate/README.md:95]: Also have to mention that we have configurable theme in theme.scss file that can be modified as per need

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: apps/boiler-plate/README.md [hunk 4]
```diff
@@ -1,27 +1,129 @@
...
+Run `npm install -g @angular/cli` for Angular CLI & NPM installed
+Run `npm i @angular/material` for Angular material
 
-Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory.
+# Code scaffolding
 
-## Running unit tests
+Run ng generate component component-name to generate a new component. You can also use ng generate directive|pipe|service|class|guard|interface|enum|module.
+
+# Project Structure
```
Comment (yeshamavani) [inline apps/boiler-plate/README.md:20]: No need of this project structure details

File: apps/boiler-plate/README.md [hunk 5]
```diff
@@ -1,27 +1,129 @@
...
+
+feature/: This directory contains feature-specific components, services, and modules for your application.
+
+# Project Features
+
+authComponent: this component authenticates the user and receives an access token from Auth
+
+commands:In the project we using forget-password,get-current-user,get-token,google-login,login,logout,refresh-token,reset-password,verify-reset-password commands to make the autheticated and secure appication
+
+loginComponent: The login component uses the authentication service to login to the application. If the user is already logged in they are automatically redirected to the home page.
```
Comment (yeshamavani) [inline apps/boiler-plate/README.md:64]: we have also included the basic put/post get etc commands that are used as the parent for other commands mention them first.

File: apps/boiler-plate/README.md [hunk 6]
```diff
@@ -1,27 +1,129 @@
...
+
+mainModule:The Main module where the User enters after login under mainModule we have homeComponent.
+
+homeComponent: The home page shows the content what is exactly going into the project
+
+adapters: In the Project adapter is a library or package that allows Angular to work with other external libraries or frameworks. Adapters help to bridge the gap between Angular and these external tools, making it easier to integrate them into an project.In project we are using logged-in-user-adapter,login-adapter
+
+decorator:In this Project decorator is a TypeScript feature used to modify or extend the behavior of a class, method, property, or parameter. Decorators in Angular are typically used to add metadata to classes or their members, which is then used by Angular's dependency injection system, routing, or other Angular features.
```
Comment (yeshamavani) [inline apps/boiler-plate/README.md:72]: No need to include the definitions instead concentrate more on what is provided.
We can just give reference links for the definitions.

File: apps/boiler-plate/README.md [hunk 7]
```diff
@@ -1,27 +1,129 @@
...
+
+Interceptors:In this project, an interceptor is a TypeScript class that can intercept HTTP requests or responses before they are sent or received by an Angular application. Interceptors are commonly used to add headers, modify requests or responses, or handle errors in a consistent way across an application.
```
Comment (yeshamavani) [inline apps/boiler-plate/README.md:74]: same

**Review by yeshamavani — CHANGES_REQUESTED** [review]
We need to focus on what we are providing rather than the definitions 
Also how a user can incorporate these features in his angular app.
What all changes will be required etc.

File: README.md [hunk 8]
```diff
@@ -1,15 +1,142 @@
...
+
+Angular Boilerplate is a starter template for Angular applications that provides various features and components to help developers get started with their projects quickly. Here are some of the features and components that the boilerplate provides:
+
+# Auth Module:
+
+This Module provides the following features:
+
+## Login Component:
+
+- login component allows users to authenticate themselves by entering their credentials, such as a username and password. The login component typically includes a form with input fields for the user's credentials, as well as a submit button to initiate the login process.
```
Comment (shubhamp-sf) [inline README.md:30]: > ... The login component typically includes ...

?
Mention what we're providing, not the general statements.

File: README.md [hunk 9]
```diff
@@ -1,15 +1,142 @@
...
+
+- login component communicates with a backend server to verify the user's credentials and grant access to the application if the user is authorized. Upon successful authentication, the login component typically stores the user's session information, such as a token or cookie, to allow the user to access protected resources without having to log in again.
```
Comment (shubhamp-sf) [inline README.md:32]: Same with this, these content seems to be the generalised statements. Include specifics of what this boilerplate have.

File: README.md [hunk 10]
```diff
@@ -1,15 +1,142 @@
...
+
+## Auth Component:
+
+Auth component handles the authentication and authorization of users. It is responsible for managing user sessions, verifying user credentials, and granting access to protected resources based on the user's role and permissions.
+
+The auth component typically includes a login as well as a registration form for new users to create an account. The component may also handle password reset functionality and provide options for users to manage their accounts.
```
Comment (shubhamp-sf) [inline README.md:38]: Same for this.

File: README.md [hunk 11]
```diff
@@ -1,15 +1,142 @@
...
+
+# Core Component:
+
+## API:
+
+- Adapters:We are using 3 adapters in this microservice named "anyAdapter,countAdapter,nameAdapter".We use these adapters for manipulating the data, as we get the data from the api.
```
Comment (shubhamp-sf) [inline README.md:44]: Microservice?
Also fix the sentence formatting like: space after the colon and dot.

File: README.md [hunk 12]
```diff
@@ -1,15 +1,142 @@
...
+
+- Commands:In this microservice we are using "del-api.command,get-api.command,get-list-api.command,patch-api.command,post-api.command,put-api.command" We are using all the commands for
+  modifying the data through API.
+
+- Models:We are using 3 models named "count.model,named-id-required.model,named-id.model" In these models we are using model based validations.
+
+## Auth:
+
+- Adaptors:Two adapter service provided are "logged-in-user-adapter.service and login-adapter.
```
Comment (shubhamp-sf) [inline README.md:53]: when mentioning the code file name use back ticks (`) not quotes, or provide a link pointing to the actual file.

File: README.md [hunk 13]
```diff
@@ -1,15 +1,142 @@
...
+  access to certain routes or pages in the application.Guards can be used to check if the user has permission to access a certain route, if the user is logged in, or to perform other types of authentication or authorization check.
+
+- Models: We are using 4 models named "logged-in-user.model,loggedin-response.model,login.model,
+  token-response.model" that represents user's data when they are logged in,response from a login API call(code),data that is required for a user to authenticate or log
+  in to a system and the the response from server after a user gets authenticated
+  and obtains a new access token and refresh token.
+
+## Decorator:
+
+In this microservice we are using 2 decorators named "required.decorator,validate.decorator",In these decorator we are providing our requirement for the data that which data needs to be fill,and than validate the data.
```
Comment (shubhamp-sf) [inline README.md:69]: Fix the grammar, formatting and spelling mistakes.

File: README.md [hunk 14]
```diff
@@ -1,15 +1,142 @@
...
+
+## Localization:
+
+In this microservice we are using i18n module as a language translator & where we are using 3 files as follows :
+
+- Enums:we are using language.enum in this microservice where this files define the set of named values.
+
+- Module:We are using localization.module while using the module we are using data from enum too and according to enums value we translate the language and call the service
+
+- Service: The translationService provides localization functionality to the application. It depends on @ngx-translate/core library to handle translations allowing users to select their preferred language for the application & to provide appropriate translations based on that preference.
```
Comment (shubhamp-sf) [inline README.md:83]: > ... It depends on @ngx-translate/core library to handle translations a ...

Use backticks or link to the npm package when specifying external packages.

File: README.md [hunk 15]
```diff
@@ -1,15 +1,142 @@
...
+
+## Env Resolver Service:
+
+The purpose of this service is to retrieve environment configuration data from a store (SystemStoreFacadeService) and make it available to components before they are displayed.
```
Comment (shubhamp-sf) [inline README.md:87]: Mention an example of how components can use env config resolved using this service.

File: README.md [hunk 16]
```diff
@@ -1,15 +1,142 @@
...
+
+## Store:
+
+In store module we have following features named store-keys.enum,user-session-store.service
```
Comment (shubhamp-sf) [inline README.md:91]: enum file is not a feature

File: README.md [hunk 17]
```diff
@@ -1,15 +1,142 @@
...
+
+## Theme:
+
+Theme module in our microservice is usually used in conjunction with Nebular, Nebular is a customizable Angular UI Library and styles designed to create a consistent, modern user interface. Nebular includes a set of pre-defined themes, but you can also create your own custom themes also.
+
+Also, provides methods to register icon packs with the NbIconLibraries service. The NbIconLibraries service is a part of the nebular UI library and is used to manage icon libraries and packs.
+
+## Toaster:
+
+In this microservice we are using toaster named Itoaster this is a user interface component that displays notifications or alerts to users in a non-intrusive way. Itoaster notifications typically appear as small pop-up messages that provide feedback or information.
```
Comment (shubhamp-sf) [inline README.md:103]: Include a code example of how someone using this boilerplate can show toasters.

**Review by shubhamp-sf — CHANGES_REQUESTED** [review]

File: README.md (hunk 11 above)
Comment (Surbhi-sharma1) [reply README.md:44]: Done

File: README.md (hunk 12 above)
Comment (Surbhi-sharma1) [reply README.md:53]: Done

File: README.md (hunk 13 above)
Comment (Surbhi-sharma1) [reply README.md:69]: Done

File: README.md (hunk 14 above)
Comment (Surbhi-sharma1) [reply README.md:83]: Done

**Pull request closed without merging**